models/POP/shell_scripts/cesm2_1/CESM2_1_setup_pmo

```diff
@@ -0,0 +1,763 @@
+# CESM_DART_config to augment the existing run script, modify STOP_N to 6
+# hours, and start harvesting synthetic observations when CESM stops at
+# 2004-02-01 06Z.
+#
+# This script relies heavily on the information in:
+# http://esmci.github.io/cime/versions/master/html/index.html
+#
+# ---------------------
+# How to use this script.
+# ---------------------
+#
+# -- You will have to read and understand the script in its entirety.
+#    You will have to modify things outside this script.
+#    This script sets up a plain CESM single-instance run without DART,
+#    intentionally.  Once it is running, calls to DART can be added.
+#
+# -- Examine the whole script to identify things to change for your experiments.
+#
+# -- Edit this script in the $DART/models/POP/shell_scripts/cesm2_1 directory
+#    or copy it to somewhere where it will be preserved.
+#
+# -- Locate the initial condition files that CESM will need.
+#
+# -- Run this script. When it is executed, it will create:
+#    1) a CESM 'CASE' directory, where the model will be built,
+#    2) a run directory, where each forecast will take place,
+#    3) a bld directory for the executables.
+#    4) The short term archiver will use a fourth directory for
+#       storage of model output until it can be moved to long term storage
+#       (campaign storage or the research data archive)
+#
+# -- To run DART; read, understand, and execute ${CASEROOT}/CESM_DART_config
+#
+# -- Submit the job using ${CASEROOT}/case.submit -M begin,end
+#
+# If you want to change something in your case other than the runtime
+# settings, it is safest to delete everything and start the run from scratch.
+#
+# ==============================================================================
+
+# ==============================================================================
+# Source DART_params.csh.
+#
+# Many of the environmental variables are stored in the DART_params.csh file
+# and are sourced by the various scripts. This increases the ease of maintenance
+# and decreases the errors associated with inconsistent script settings.
+# ==============================================================================
+
+echo ''
+echo 'Source DART_params.csh.'
+echo ''
+
+source ./DART_params.csh || exit 1
+
+# ==============================================================================
+# Set CASE to the combination of the compset, resolution and an optional extra
+# string.
+#
+# extra_string  An optional extra_string to differentiate a case from others
+#               that have the same compset and resolution.
+# CASE          The value of "case" will be used many ways; directory and file
+#               names both locally and on HPSS, and script names; so consider
+#               its length and information content.
+# ==============================================================================
+
+setenv extra_string ''
+setenv CASE         ${compset}.${resolution}${extra_string}
```

Comment:
I am not sure about this strategy ... it only works because you've chosen to use the shortest possible definition of a very specific compset ... 'G' ... this is not necessarily how it will be used in general.   In CLM and CAM, we choose to (have to?) specify the compset longname.  Your 'G' resolves to '2000_DATM%NYF_SLND_CICE_POP2_DROF%NYF_SGLC_WW3'
which is kinda awful to be part of a casename.


Compset longname is 2000_DATM%NYF_SLND_CICE_POP2_DROF%NYF_SGLC_WW3
Compset specification file is /glade/u/home/cmip6/cesm_tags/release-cesm2.1.1/cime/../components/pop//cime_config/config_compsets.xml
Compset forcing is 1972-2004
ATM component is  Data driven ATM COREv2 normal year forcing
LND component is Stub land component
ICE component is Sea ICE (cice) model version 5
OCN component is POP2 
ROF component is Data runoff modelCOREv2 normal year forcing:
GLC component is Stub glacier (land ice) component
WAV component is Wave Watch

Reply:
Added an additional environmental variable `case_string` in `DART_params`.